### PR TITLE
Assorted fixes

### PR DESCRIPTION
**Pull request by lucasgelfond**
Docs changes, youtube-dl --> yt-dlp, suggestions for Apple Silicon. 

Also generally have meta question about how we want to do PRs here. My instinct is - lots of small changes vs. one big one. Problem is that each individual small change tends to be breaking (even if getting closer to an outcome that works), so possible certain changes break further. 

For example, my workflow of using `./mtriage dev` and calling `python3 src/run.py` is failing (I get a 'Illegal instruction' which is a Python error I've never seen before) and the general `./mtriage run docs/tutorial/3/3b.yaml` is running (better than I got before!) but seems to just hang on the command (i.e., I'm just stuck here):
Screenshot 2023-05-01 at 10 17 36 AM

Lmk what you prefer! Happy to do whatever but figured good to start the process..

### CODE REVIEW

**Comment by lucasgelfond** [comment]
cc @breezykermo

**Comment by lucasgelfond** [comment]
Maybe a better way to say this actually is - the `src/run.py` workflow works as I was using it before (i.e., for 3a in the tutorial), but not for later things we're still working through, like 3b

**Comment by lucasgelfond** [comment]
also cc @jackson-delea

**Comment by lucasgelfond** [comment]
This also works now when I try to build a new dev Docker image, once we get some more testing in possible we also want to publish this in place of current image?
Screenshot 2023-05-01 at 10 46 09 AM

File: docs/install.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ Ubuntu installation [here](https://docs.docker.com/v17.12/install/linux/docker-c
 If you have a CUDA GPU, you can use [Nvidia Docker](https://github.com/NVIDIA/nvidia-docker)
 instead of Docker to make certain analysers more performant.
 
+NOTE (05/2023): if you are on Apple Silicon, your machine will not be able to natively run FA's docker images. In order to fix this, you'll need to [enable virtualization](https://collabnix.com/warning-the-requested-images-platform-linux-amd64-does-not-match-the-detected-host-platform-linux-arm64-v8/) by changing some settings in Docker Desktop. Navigate to Settings > General and make sure the "Use Virtualization framework" box is checked. After, navigate to Settings > Features in development and check the "Use Rosetta for x86/amd64 emulation on Apple Silicon." If you're well-versed in Docker, you can set the 'platform' flag to 'linux/amd64' in the Dockerfile. If not, the easiest solution is to modify your personal ~/.bashrc or ~/.zshrc file and add ``export DOCKER_DEFAULT_PLATFORM=linux/amd64`` to it.
```
Comment (breezykermo) [inline docs/install.md]: Thanks, this is a good comment. We could also look into building a Docker image that is compatible with Apple Silicon.

File: docs/tutorial/1/README.md [hunk 2]
```diff
@@ -34,8 +34,8 @@ You should see the following logs:
 ```
 Local: index: Indexing local folder...
 Local: index: indexed file: 1.txt
-Local: index: indexed file: 3.jpg
 Local: index: indexed file: 2.md
+Local: index: indexed file: 3.jpg
```
Comment (breezykermo) [inline docs/tutorial/1/README.md]: I'd prefer to leave this order in the docs as is, as it is actually representing the fact that mtriage can run analysis/selection of different elements in parallel (and thus the order of logs here is not strictly sequential).

File: docs/tutorial/1/README.md [hunk 3]
```diff
@@ -95,8 +95,8 @@ We should see the following logs:
 ```
 ExtractTypes: None: Running in parallel
 ExtractTypes: analyse: Extracting element 1 with paths: ['1.txt']
-ExtractTypes: analyse: No extracted media in element 3.
 ExtractTypes: analyse: Extracting element 2 with paths: ['2.md']
+ExtractTypes: analyse: No extracted media in element 3.
```
Comment (breezykermo) [inline docs/tutorial/1/README.md]: Likewise here, see comment above.

File: docs/tutorial/3/3a.yaml [hunk 4]
```diff
@@ -3,5 +3,5 @@ select:
   name: Youtube
   config:
     search_term: Tear gas
-    uploaded_before: "2015-10-02T00:00:00Z"
-    uploaded_after: "2015-10-01T00:00:00Z"
+    uploaded_before: "2023-01-02T00:00:00Z"
+    uploaded_after: "2023-01-01T00:00:00Z"
```
Comment (breezykermo) [inline docs/tutorial/3/3a.yaml]: Does the 2015 search no longer work? This shouldn't necessarily be pegged to a more recent date; but it should return at least some results so that the analysis pipeline is meaningful.

File: docs/tutorial/3/README.md [hunk 5]
```diff
@@ -31,6 +31,14 @@ available. By default, it downloads the videos at very low quality, and it also
...
 information.
 
+
+NOTE: This step currently doesn't work, and, until we get a better Docker image up, you'll need to use a workaround. The installation as-is uses an old version of youtube-dl and cannot properly fetch videos. To fix this, you need to:
+
+1. Run ``./mtriage dev --yaml docs/tutorial/3/3a.yaml`` to open a shell in the terminal with this configuration.
+2. Run ``pip3 install yt-dlp``. It may give you an error message, but it's likely this is okay.
+3. Check that yt-dlp has installed correctly. Run ``pip3 show yt-dlp`` and you should see a version installed. (Also okay if this command fails).
+4. Run ``python3 src/run.py`` which should work properly.
+
```
Comment (breezykermo) [inline docs/tutorial/3/README.md]: Instead of adding this to the README, I would prefer to simply update the mtriage dependencies, Docker image and youtube selector to use yt-dlp.

It looks below like you've basically done this. Is there anything stopping us from just updating the official image in Docker hub with this new build from your perspective? (I know that neither the process of building a new mtriage image, nor of uploading it to Docker Hub manually or through CI, is documented: so this is something we can do together.)

File: whitelist.txt [hunk 6]
```diff
@@ -0,0 +1,4 @@
+Youtube
+Frames
+ImageDedup
+KerasPretrained
```
Comment (breezykermo) [inline whitelist.txt]: This whitelist file shouldn't be included in the commit.

**Review by breezykermo — CHANGES_REQUESTED** [review]
Thanks for this PR! It's a great start. I've made some comments inline, and at a high level what I'd suggest is splitting this into two PRs:
1. A PR with the general fixes reflected in this PR (that I haven't commented on directly as they seem correct to me).
2. A PR that replaces youtube-dl with yt-dlp.

Does that make sense?
